src/main/java/teammates/storage/api/FeedbackSessionsDb.java

```diff
@@ -82,6 +83,42 @@ public FeedbackSessionAttributes getFeedbackSession(String courseId, String feed
         return feedbackSession;
     }
 
+    /**
+     * Gets a list of feedback sessions within the given time range.
+     */
+    public List<FeedbackSessionAttributes> getAllFeedbackSeesionWithinTimeRange(Instant rangeStart, Instant rangeEnd) {
+
+        List<FeedbackSessionAttributes> lst = new LinkedList<>();
+
+        List<FeedbackSession> startEntities = load()
+                .filter("startTime >=", rangeStart)
+                .filter("startTime <", rangeEnd)
+                .list();
+        List<FeedbackSession> endEntities = load()
+                .filter("endTime >=", rangeStart)
+                .filter("endTime <", rangeEnd)
+                .list();
+
+        List<FeedbackSession> endTimeEntities = new ArrayList<>(endEntities);
+        List<FeedbackSession> startTimeEntities = new ArrayList<>(startEntities);
+
+        endTimeEntities.removeAll(startTimeEntities);
+
+        List<FeedbackSession> resultsVisibleEntities = load()
+                .filter("resultsVisibleFromTime >", rangeStart)
+                .filter("resultsVisibleFromTime <=", rangeEnd)
+                .list();
+
+        resultsVisibleEntities.removeAll(startEntities);
+        resultsVisibleEntities.removeAll(endEntities);
+
+        lst.addAll(makeAttributes(startEntities));
+        lst.addAll(makeAttributes(endEntities));
+        lst.addAll(makeAttributes(resultsVisibleEntities));
+
+        return lst;
```

Comment:
what about we order them a bit? e.g. according to start time or name or deadlines...